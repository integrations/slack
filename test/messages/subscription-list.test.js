const SubscriptionList = require('../../lib/messages/subscription-list');

describe('SubscriptionList', () => {
  test('works for multiple subscriptions active in a channel', async () => {
    const repositories = [
      {
        full_name: 'bkeepers/dotenv',
        html_url: 'https://github.com/bkeepers/dotenv',
      },
      {
        full_name: 'atom/atom',
        html_url: 'https://github.com/atom/atom',
      },
    ];
    const organizations = [
      {
        login: 'kubernetes',
        html_url: 'https://github.com/kubernetes',
      },
      {
        login: 'atom',
        html_url: 'https://github.com/atom',
      },
    ];
    expect(new SubscriptionList(repositories, organizations, 'C01234').toJSON()).toMatchSnapshot();
  });

  test('works for one subscription active in a channel', async () => {
    const repositories = [
      {
        full_name: 'bkeepers/dotenv',
        html_url: 'https://github.com/bkeepers/dotenv',
      },
    ];
    const organizations = [
      {
        login: 'kubernetes',
        html_url: 'https://github.com/kubernetes',
      },
    ];
    expect(new SubscriptionList(repositories, organizations, 'C01234').toJSON()).toMatchSnapshot();
  });

  test('works for no subscriptions active in a channel', async () => {
    const repositories = [];
    const organizations = [];
    expect(new SubscriptionList(repositories, organizations, 'C01234').toJSON()).toMatchSnapshot();
  });

  test('works for one subscription active in a direct message', async () => {
    const repositories = [
      {
        full_name: 'bkeepers/dotenv',
        html_url: 'https://github.com/bkeepers/dotenv',
      },
    ];
    const organizations = [
      {
        login: 'kubernetes',
        html_url: 'https://github.com/kubernetes',
      },
    ];
    expect(new SubscriptionList(repositories, organizations, 'D01234').toJSON()).toMatchSnapshot();
  });

  test('sorts subscriptions alphabetically', async () => {
    const repositories = [
      {
        full_name: 'wilhelmklopp/wilhelmklopp',
        html_url: 'https://github.com/wilhelmklopp/wilhelmklopp',
      },
      {
        full_name: 'bkeepers/dotenv',
        html_url: 'https://github.com/bkeepers/dotenv',
      },
      {
        full_name: 'integrations/slack',
        html_url: 'https://github.com/integrations/slack',
      },
    ];
    expect(new SubscriptionList(repositories, [], 'C01234').repositoriesToString()).toEqual([
      '<https://github.com/bkeepers/dotenv|bkeepers/dotenv>',
      '<https://github.com/integrations/slack|integrations/slack>',
      '<https://github.com/wilhelmklopp/wilhelmklopp|wilhelmklopp/wilhelmklopp>',
    ]);
  });

  test('sorts repositories alphabetically including different cases', async () => {
    const repositories = [
      {
        full_name: 'wilhelmklopp/wilhelmklopp',
        html_url: 'https://github.com/wilhelmklopp/wilhelmklopp',
      },
      {
        full_name: 'bkeepers/dotenv',
        html_url: 'https://github.com/bkeepers/dotenv',
      },
      {
        full_name: 'integrations/slack',
        html_url: 'https://github.com/integrations/slack',
      },
      {
        full_name: 'JasonEtco/todo',
        html_url: 'https://github.com/JasonEtco/todo',
      },
    ];
    expect(new SubscriptionList(repositories, [], 'C01234').repositoriesToString()).toEqual([
      '<https://github.com/bkeepers/dotenv|bkeepers/dotenv>',
      '<https://github.com/integrations/slack|integrations/slack>',
      '<https://github.com/JasonEtco/todo|JasonEtco/todo>',
      '<https://github.com/wilhelmklopp/wilhelmklopp|wilhelmklopp/wilhelmklopp>',
    ]);
  });

  test('sorts accounts alphabetically', async () => {
    const organizations = [
      {
        login: 'kubernetes',
        html_url: 'https://github.com/kubernetes',
      },
      {
        login: 'atom',
        html_url: 'https://github.com/atom',
      },
    ];
    expect(new SubscriptionList([], organizations, 'C01234').accountsToString()).toEqual([
      '<https://github.com/atom|atom>',
      '<https://github.com/kubernetes|kubernetes>',
    ]);
  });

  test('sorts accounts alphabetically including different cases', async () => {
    const organizations = [
      {
        login: 'Microsoft',
        html_url: 'https://github.com/Microsoft',
      },
      {
        login: 'kubernetes',
        html_url: 'https://github.com/kubernetes',
      },
      {
        login: 'atom',
        html_url: 'https://github.com/atom',
      },
    ];
    expect(new SubscriptionList([], organizations, 'C01234').accountsToString()).toEqual([
      '<https://github.com/atom|atom>',
      '<https://github.com/kubernetes|kubernetes>',
      '<https://github.com/Microsoft|Microsoft>',
    ]);
  });
});
