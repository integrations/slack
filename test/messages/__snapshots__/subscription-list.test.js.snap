// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubscriptionList works for multiple subscriptions active in a channel 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 2 repositories",
      "text": "<https://github.com/atom/atom|atom/atom>
<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
      "title": "Subscribed to the following repositories",
    },
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 2 accounts",
      "text": "<https://github.com/atom|atom>
<https://github.com/kubernetes|kubernetes>",
      "title": "Subscribed to the following accounts",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscriptionList works for no subscriptions active in a channel 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "Not subscribed to any repositories or accounts",
      "title": "Not subscribed to any repositories or accounts",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscriptionList works for one subscription active in a channel 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 1 repository",
      "text": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
      "title": "Subscribed to the following repositories",
    },
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 1 account",
      "text": "<https://github.com/kubernetes|kubernetes>",
      "title": "Subscribed to the following accounts",
    },
  ],
  "response_type": "in_channel",
}
`;

exports[`SubscriptionList works for one subscription active in a direct message 1`] = `
Object {
  "attachments": Array [
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 1 repository",
      "text": "<https://github.com/bkeepers/dotenv|bkeepers/dotenv>",
      "title": "Subscribed to the following repositories",
    },
    Object {
      "color": "#24292f",
      "fallback": "Subscribed to 1 account",
      "text": "<https://github.com/kubernetes|kubernetes>",
      "title": "Subscribed to the following accounts",
    },
  ],
  "response_type": "in_channel",
}
`;
